assemble.go

```diff
@@ -128,6 +128,22 @@ func AssembleFile(ctx context.Context, name string, idx Index, s Store, seeds []
 					if err != nil {
 						return err
 					}
+
+					// Validate that the written chunks are exactly what we were expecting.
+					// Because the seed might point to a RW location, if the data changed
+					// while we were extracting an index, we might end up writing to the
+					// destination some unexpected values.
+					for _, c := range job.segment.chunks() {
+						b := make([]byte, c.Size)
+						if _, err := f.ReadAt(b, int64(c.Start)); err != nil {
+							return err
+						}
+						sum := Digest.Sum(b)
+						if sum != c.ID {
+							return fmt.Errorf("written data in %s doesn't match its expected value", name)
```

Comment:
Since at this point we can be pretty sure that a seed was modified during processing, we could be more clear in the message to help anyone that sees this message and debugging it. How about this?

`written data in %s doesn't match its expected hash value, seed may have changed during processing`

Reply:
Right, I changed the error message with your suggested string.